htdocs/core/lib/functions.lib.php

```diff
@@ -5544,9 +5544,9 @@ function getCommonSubstitutionArray($outputlangs, $onlykey=0, $exclude=null, $ob
 	{
 		$substitutionarray['__DATE_YMD__']        = is_object($object)?(isset($object->date) ? dol_print_date($object->date, 'day', 0, $outputlangs) : '') : '';
 		$substitutionarray['__DATE_DUE_YMD__']    = is_object($object)?(isset($object->date_lim_reglement)? dol_print_date($object->date_lim_reglement, 'day', 0, $outputlangs) : '') : '';
-		$substitutionarray['__AMOUNT__']          = is_object($object)?$object->total_ttc:'';
-		$substitutionarray['__AMOUNT_EXCL_TAX__'] = is_object($object)?$object->total_ht:'';
-		$substitutionarray['__AMOUNT_VAT__']      = is_object($object)?($object->total_vat?$object->total_vat:$object->total_tva):'';
+		$substitutionarray['__AMOUNT__']          = is_object($object)?price($object->total_ttc):'';
```

Comment:
The "price" function is a function that format the date for output (according to user language and decimal setup). So original value of amout is corrupted and can not be used anymore in a template that make calculation for example.

As a solution, i backported the new substitution keys that were intorduced in v8
__AMOUNT_FORMATED__
__AMOUNT_EXCL_TAX_FORMATED__
__AMOUNT_TAX_FORMATED__
that will have content of price function.

Reply:
Ok, I will use these keys
Thanks